util-cache/src/main/contraband/remotecache.contra

```diff
@@ -0,0 +1,16 @@
+package sbt.util
+@target(Scala)
+
+type UpdateActionResultRequest {
+  actionDigest: sbt.util.Digest!
+  outputFiles: [xsbti.VirtualFile] @since("0.1.0")
+  exitCode: Int @since("0.2.0")
+  isExecutable: [Boolean] @since("0.3.0")
+}
+
+type GetActionResultRequest {
+  actionDigest: sbt.util.Digest!
+  inlineStdout: Boolean @since("0.1.0")
+  inlineStderr: Boolean @since("0.1.0")
+  inlineOutputFiles: [String] @since("0.2.0")
```

Comment:
What is this `inlineOutputFiles`? What does it mean?

Reply:
This is from https://github.com/bazelbuild/remote-apis/blob/96942a2107c702ed3ca4a664f7eeb7c85ba8dc77/build/bazel/remote/execution/v2/remote_execution.proto#L1582:
> A hint to the server to inline the contents of the listed output files.

Using our `ActionCacheStore` lingo, to get a task result even for Int `1`, we have to make two roundtrip calls: `ActionCacheStore#get` to get the reference to CAS entry of JSON, and `ActionCacheStore#syncBlobs` to download JSON from CAS. But it is inefficient in multiple ways:
- Second roundtrip just for `1`
- File creation on the laptop just for `1`

So `GetActionResultRequest` has an optimization that can optionally piggyback the content of "outputFile" as bytearray encoded into gRPC response. If the blob is too big, the server has the right to ignore it, but for most task results we should ask to inline, and check if the result exists, which is what `ActionCache.cache` does now:

```scala
        // some protocol can embed values into the result
        result.contents.headOption match
```